src/System.Runtime.Extensions/src/System/Runtime/Versioning/FrameworkName.cs

```diff
@@ -62,19 +62,25 @@ public String FullName
             {
                 if (_fullName == null)
                 {
-                    StringBuilder sb = new StringBuilder();
-                    sb.Append(Identifier);
-                    sb.Append(c_componentSeparator);
-                    sb.Append(c_versionKey).Append(c_keyValueSeparator);
-                    sb.Append(c_versionValuePrefix);
-                    sb.Append(Version);
-                    if (!String.IsNullOrEmpty(Profile))
+                    if (string.IsNullOrEmpty(Profile))
                     {
-                        sb.Append(c_componentSeparator);
-                        sb.Append(c_profileKey).Append(c_keyValueSeparator);
-                        sb.Append(Profile);
+                        // Roslyn will optimize this to call string.Concat, avoiding any intermediary allocations
+                        _fullName = Identifier + c_componentSeparator + c_versionKey + c_keyValueSeparator + c_versionValuePrefix + Version;
```

Comment:
Please change ```Version``` to ```Version.ToString()```.  Doing so in this case won't save any allocations, but it will make the call slightly faster.  As currently written, this will use ```string.Concat(object, object, object)```, which needs to do null checks, call ToString on each input, and then call ```string.Concat(string, string, string)```... we can just get the compiler to call the latter directly by calling ToString on the Version ourselves such that all of the three inputs are strings.